src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -844,7 +844,7 @@ private ClusteringIndexFilter makeFilter(List<PrimaryKey> keys)
         "PrimaryKey " + firstKey + " clustering does not match table. There should be a clustering of size " + cfs.metadata().comparator.size();
 
         ClusteringIndexFilter clusteringIndexFilter = command.clusteringIndexFilter(firstKey.partitionKey());
-        if (cfs.metadata().comparator.size() == 0 || firstKey.hasEmptyClustering())
+        if (cfs.metadata().comparator.size() == 0 || firstKey.hasEmptyClustering() || !indexFeatureSet.isRowAware())
```

Comment:
Do I understand it correctly that essentially switches the whole query execution to non-row-aware mode even if only one AA index exists in the view? We're not longer fetching individual rows, but full partitions only.